src/middlewared/middlewared/plugins/cluster_linux/ctdb_private.py

```diff
@@ -0,0 +1,241 @@
+from middlewared.schema import Dict, IPAddr
+from middlewared.service import (accepts, private, job,
+                                 Service, ValidationErrors,
+                                 CallError)
+from middlewared.plugins.cluster_linux.utils import CTDBConfig
+import pathlib
+
+
+SHARED_VOL = CTDBConfig.CTDB_LOCAL_MOUNT.value
+PRI_IP_FILE = CTDBConfig.GM_PRI_IP_FILE.value
+ETC_IP_FILE = CTDBConfig.ETC_PRI_IP_FILE.value
+PRI_LOCK = CTDBConfig.PRI_LOCK.value
+
+
+class CtdbIpService(Service):
```

Comment:
I would probably make this a CRUDService

Reply:
I didn't make it a CRUDService because I'm not writing anything to the db?

Reply:
Its not about writing to the db or not, but consistent API.

Also, are these really supposed to be both public APIs?

Reply:
Okay well I'm being told 2 different things then. I've been told not to make it CRUD if it doesn't touch our dB. That's the entire approach I took when writing the gluster api

Reply:
Was I the one who told you that? I think there was some miscommunication. CRUD was never related to DB.

Reply:
The questions I have now are:

1. When do I not use CRUDService?
2. Do you want me to back and fix gluster plugin?

Reply:
You do not use CRUDService when it does not make sense to have a query method, nor a create/delete method.

CRUDService is for when you're operating over a single entity:

CRUD - Create/Retrieve/Update/Delete
Create - A single entity (e.g. an ip, an interface, a user)
Retrieve - A collection (a list of objects)
Update - A single entity (e.g. in the interface X, set attribute Y to Z)
Delete - A single entity (e.g. delete an interface X)

Reply:
About fixing gluster, if that would make things consistent, I am +1 for that, as that API is not considered stable yet

Reply:
Okay thanks, I'll take care of it.